Examples/Examples/All Examples/AnimateGeoJSONLine.swift

```diff
@@ -0,0 +1,189 @@
+                                        zoom: 11.0)
+        
+        allCoordinates = coordinates
+        
+        // Allows the delegate to receive information about map events.
+        mapView.on(.mapLoadingFinished) { [weak self] _ in
+            
+            guard let self = self else { return }
+            self.addLine()
+            self.animatePolyline()
+            
+            // The below line is used for internal testing purposes only.
+             self.finish()
+        }
+    }
+    
+    func addLine() {
+        let coordinate = coordinates[currentIndex]
+        // Create a GeoJSON data source.
+        geoJSONLine.source.data = .feature(Feature(geometry: Geometry.lineString(LineString([coordinate]))))
+        
+        var lineLayer = LineLayer(id: "line-layer")
+        lineLayer.filter = Exp(.eq) {
+            "$type"
+            "LineString"
+        }
+        lineLayer.source = geoJSONLine.identifier
+        lineLayer.paint?.lineColor = .constant(ColorRepresentable(color: UIColor.red))
+        
+        let lowZoomWidth = 5
+        let highZoomWidth = 20
+        lineLayer.paint?.lineWidth = .expression(
+            Exp(.interpolate) {
+                Exp(.linear)
+                Exp(.zoom)
+                14
+                lowZoomWidth
+                18
+                highZoomWidth
+            }
+        )
+        lineLayer.layout?.lineCap = .round
+        lineLayer.layout?.lineJoin = .round
+        
+        // Add the lineLayer to the map.
+        self.mapView.style.addSource(source: geoJSONLine.source, identifier: geoJSONLine.identifier)
+        self.mapView.style.addLayer(layer: lineLayer)
+        
+    }
+    
+    func animatePolyline() {
+        currentIndex = 1
+        
+        // Start a timer that will simulate adding points to our polyline. This could also represent coordinates being added to our polyline from another source, such as a CLLocationManagerDelegate.
+        timer = Timer.scheduledTimer(timeInterval: 0.10, target: self, selector: #selector(tick), userInfo: nil, repeats: true)
+    }
+    
+    @objc func tick() {
+        if currentIndex > allCoordinates.count {
+            timer?.invalidate()
+            timer = nil
+            return
+        }
+        
+        // Create a subarray of locations up to the current index.
+        let coordinates = Array(allCoordinates[0..<currentIndex])
+        
+        // Update our lineLayer with the current locations.
+        updatePolylineWithCoordinates(coordinates: coordinates)
+        
+        currentIndex += 1
+        print(currentIndex)
+    }
+    
+    func updatePolylineWithCoordinates(coordinates: [CLLocationCoordinate2D]) {
+        let mutableCoordinates = coordinates
+
+        // Identify the new coordinate to animate to
+        geoJSONLine.source.data = .feature(Feature(LineString(mutableCoordinates)))
+        let geoJSON = Feature.init(geometry: Geometry.lineString(LineString(mutableCoordinates)))
```

Comment:
We added an extension to `Feature` you can use to avoid the long initializer - you can now turn:

```swift
Feature.init(geometry: Geometry.lineString(LineString(mutableCoordinates)))
```

Into this:

```swift
Feature(LineString(mutableCoordinates)
```